### PR TITLE
[TT-10358] Add fallback to default version OAS support

### PR DESCRIPTION
**Pull request by furkansenharputlu**
parent ticket: https://tyktech.atlassian.net/browse/TT-9901
subticket: https://tyktech.atlassian.net/browse/TT-10358

### CODE REVIEW

File: apidef/oas/root.go [hunk 1]
```diff
@@ -190,6 +192,7 @@ func (v *Versioning) Fill(api apidef.APIDefinition) {
 	}
 
 	v.StripVersioningData = api.VersionDefinition.StripVersioningData
+	v.FallbackToDefault = api.VersionDefinition.StripVersioningData
```
Comment (jeffy-mathew) [inline apidef/oas/root.go:195]:
```suggestion
	v.FallbackToDefault = api.VersionDefinition.FallbackToDefault
```

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5679/merge
Commit: 1f1f3e3d562bcbeefe6afdb347e8c667d2c74140 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6653486683)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5679/merge
Commit: 17707a876b7d57a17571619e1da2e0c2c4eeb269 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6653830569)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5679/merge
Commit: 39efb841ddec71dadb79c22adbd8b7531f6ae26d 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6653840130)

File: apidef/oas/schema/x-tyk-gateway.md [hunk 2]
```diff
@@ -88,6 +88,9 @@ Versions contains a list of versions that map to individual API IDs.
 **Field: `stripVersioningData` (`boolean`)**
 StripVersioningData is a boolean flag, if set to `true`, the API responses will be stripped of versioning data.
 
+**Field: `fallbackToDefault` (`boolean`)**
+FallbackToDefault controls whether the default API is invoked when the requested version can't be found.
```
Comment (titpetric) [inline apidef/oas/schema/x-tyk-gateway.md:92]:
```suggestion
FallbackToDefault is a boolean flag, if set to `true`, the default API is invoked when the requested version can't be found.
```

If it's good enough for `stripVersioningData`, it's good enough here

Comment (furkansenharputlu) [reply apidef/oas/schema/x-tyk-gateway.md:92]: It starts like `**Field: fallbackToDefault (boolean)**`. Why would we need to say `a boolean` flag again. I think `stripVersioningData` should be changed here.

Comment (andyo-tyk) [reply apidef/oas/schema/x-tyk-gateway.md:92]: I agree with @furkansenharputlu that this is superfluous. The existing docs [page](https://tyk.io/docs/tyk-apis/tyk-gateway-api/oas/x-tyk-oas-doc/) has a mix of styles for describing Booleans, though most don't have this phrase.

My suggested wording is:

 ```suggestion
FallbackToDefault controls the behaviour of Tyk when a versioned API is called with a nonexistent version number. If set to `true` then the default API version will be invoked; if set to `false` Tyk will return an HTTP 404 "Version does not exist" error in this scenario.
```

**Review by andyo-tyk — CHANGES_REQUESTED** [review]
Suggested rewording of the GoDocs comment

**Review by jeffy-mathew — APPROVED** [review]

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5679/merge
Commit: f138f07fb5ca2e191f5a3682d9da8634cf4df8de 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6689456238)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5679/merge
Commit: 99a531b0513c05e5ba36054ed605b89012168745 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6689442364)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5679/merge
Commit: 576fa89b4bf8465b20d77112687d4e87c982b75b 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6689508823)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5679/merge
Commit: 3430ca4251561f403079a40fd2eb9a199a630ab5 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6689523607)

**Review by andyo-tyk — APPROVED** [review]
Thanks - looks good.

**Pull request merged**